File: test-case/floating/matrix/Matrix.cs
using Soda.Unittest;

public class Solution {
    public double[][] MatrixMultiply(double[][] a, IList<IList<double>> b)
    {
        int rows = a.Length, cols = b[0].Count;
        var res = new double[rows][];
        for (int i = 0; i < rows; ++i) {
            res[i] = new double[cols];
            for (int j = 0; j < cols; ++j) {
                double c = 0.0;
                for (int k = 0; k < b.Count; ++k) {
                    c += a[i][k] * b[k][j];
                }
                res[i][j] = c;
            }
        }
        return res;
    }
}

public class Matrix
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().MatrixMultiply));
        // work.SetValidator((e, r) => ...);
        // work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}

File: test-case/floating/matrix/Matrix.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

class Solution {}
object Solution {
  def matrixMultiply(a: Array[Array[Double]], b: Array[Array[Double]]): Array[Array[Double]] = {
    val rows = a.size
    val cols = b(0).size
    val res = Array.ofDim[Double](rows, cols)
    for (i <- 0 until rows) {
      for (j <- 0 until cols) {
        var c: Double = 0.0
        for (k <- b.indices) {
          c += a(i)(k) * b(k)(j)
        }
        res(i)(j) = c
      }
    }
    res
  }
}

class Matrix extends (String => String) {
  override def apply(text: String): String = {
    val work = TestWork.forObject(typeOf[Solution], "matrixMultiply")
    // val work = TestWork.forStruct(...)
    // work.setValidator((R, R) => Boolean)
    // work.compareSerial = true
    work.run(text)
  }
}

object Matrix extends App {
  println(new Matrix()(Utils.fromStdin()))
}
